resources/splunk_service.rb

```diff
@@ -62,14 +65,30 @@ def install_state
   action_class do
     include CernerSplunk::ProviderHelpers
 
-    def service_action(desired_action)
+    # Provide file and service resources from methods to eliminate accidental resource cloning
+    def marker_resource
+      resources(file: marker_path.to_s)
+    rescue Chef::Exceptions::ResourceNotFound
+      file(marker_path.to_s) do
+        action :nothing
+      end
+    end
+
+    def service_resource
```

Comment:
Why do we have this indirection? Would it be more straightfoward to just create the resources in each action to do the right things/notifications? (Shouldn't they be a forked run so not worried about cloning? Or is that not true?)